webapp/channels/src/components/team_settings/team_info_tab/team_info_tab.tsx

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import type {ChangeEvent} from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+
+import type {Team} from '@mattermost/types/teams';
+
+import type {BaseSettingItemProps} from 'components/widgets/modals/components/base_setting_item';
+import ModalSection from 'components/widgets/modals/components/modal_section';
+import SaveChangesPanel, {type SaveChangesPanelState} from 'components/widgets/modals/components/save_changes_panel';
+
+import Constants from 'utils/constants';
+
+import TeamDescriptionSection from './team_description_section';
+import TeamNameSection from './team_name_section';
+import TeamPictureSection from './team_picture_section';
+
+import type {PropsFromRedux, OwnProps} from '.';
+
+import './team_info_tab.scss';
+
+const ACCEPTED_TEAM_IMAGE_TYPES = ['image/jpeg', 'image/png', 'image/bmp'];
+const translations = defineMessages({
+    Required: {
+        id: 'general_tab.required',
+        defaultMessage: 'This field is required',
+    },
+    TeamNameRestrictions: {
+        id: 'general_tab.teamNameRestrictions',
+        defaultMessage: 'Team Name must be {min} or more characters up to a maximum of {max}. You can add a longer team description.',
+        values: {min: Constants.MIN_TEAMNAME_LENGTH, max: Constants.MAX_TEAMNAME_LENGTH},
+    },
+    TeamIconInvalidFileType: {
+        id: 'general_tab.teamIconInvalidFileType',
+        defaultMessage: 'Only BMP, JPG or PNG images may be used for team icons',
+    },
+    TeamIconTooLarge: {
+        id: 'general_tab.teamIconTooLarge',
+        defaultMessage: 'Unable to upload team icon. File is too large.',
+    },
+    TeamIconError: {
+        id: 'general_tab.teamIconError',
+        defaultMessage: 'An error occurred while selecting the image.',
+    },
+});
+type Props = PropsFromRedux & OwnProps;
+
+const InfoTab = ({team, hasChanges, maxFileSize, closeModal, collapseModal, hasChangeTabError, setHasChangeTabError, setHasChanges, actions}: Props) => {
```

Comment:
Same as with the other index file. We can get maxFileSize with `useSelector` and use the actions directly with `useDispatch`.